[Bug]: telegram-bot-api build failed ndk25

### Problem description

/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:17376:7: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 3 were provided
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:17390:3: note: candidate constructor not viable: requires 0 arguments, but 3 were provided
  inputSticker();
  ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:77:31: error: no matching constructor for initialization of 'td::td_api::invoice'
  return object_ptr<Type>(new Type(std::forward<Args>(args)...));
                              ^    ~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/.termux-build/telegram-bot-api/src/telegram-bot-api/Client.cpp:7112:23: note: in instantiation of function template specialization 'td::td_api::make_object<td::td_api::invoice, std::basic_string<char>, std::vector<td::tl::unique_ptr<td::td_api::labeledPricePart>>, long &, std::vector<long>, bool, bool &, bool &, bool &, bool &, bool &, bool &, bool &>' requested here
                      make_object<td_api::invoice>(
                      ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:18706:3: note: candidate constructor not viable: requires 13 arguments, but 12 were provided
  invoice(string const &currency_, array<object_ptr<labeledPricePart>> &&price_parts_, int53 max_tip_amount_, array<int53> &&suggested_tip_amounts_, string const &recurring_payment_terms_of_service_url_, bool is_test_, bool need_name_, bool need_phone_number_, bool need_email_address_, bool need_shipping_address_, bool send_phone_number_to_provider_, bool send_email_address_to_provider_, bool is_flexible_);
  ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:18655:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 12 were provided
class invoice final : public Object {
      ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:18655:7: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 12 were provided
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:18687:3: note: candidate constructor not viable: requires 0 arguments, but 12 were provided
  invoice();
  ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:77:31: error: no matching constructor for initialization of 'td::td_api::createNewStickerSet'
  return object_ptr<Type>(new Type(std::forward<Args>(args)...));
                              ^    ~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/.termux-build/telegram-bot-api/src/telegram-bot-api/Client.cpp:8222:29: note: in instantiation of function template specialization 'td::td_api::make_object<td::td_api::createNewStickerSet, long &, std::basic_string<char>, std::basic_string<char>, std::vector<td::tl::unique_ptr<td::td_api::inputSticker>>, std::basic_string<char>>' requested here
               send_request(make_object<td_api::createNewStickerSet>(user_id, title.str(), name.str(),
                            ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:47667:3: note: candidate constructor not viable: requires 6 arguments, but 5 were provided
  createNewStickerSet(int53 user_id_, string const &title_, string const &name_, object_ptr<StickerType> &&sticker_type_, array<object_ptr<inputSticker>> &&stickers_, string const &source_);
  ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:47633:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 5 were provided
class createNewStickerSet final : public Function {
      ^
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:47633:7: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 5 were provided
/home/builder/.termux-build/telegram-bot-api/src/td/td/generate/auto/td/telegram/td_api.h:47653:3: note: candidate constructor not viable: requires 0 arguments, but 5 were provided
  createNewStickerSet();
  ^
16 errors generated.
[22/322] Building CXX object CMakeFiles/telegram-bot-api.dir/telegram-bot-api/telegram-bot-api.cpp.o
ninja: build stopped: subcommand failed.

[telegram-bot-api.txt](https://github.com/termux/termux-packages/files/9333814/telegram-bot-api.txt)


### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

termux-info:

```

```

--- a/packages/telegram-bot-api/build.sh
+++ b/packages/telegram-bot-api/build.sh
@@ -7,18 +7,19 @@ _COMMIT=8a0f1dd730aa41ab7b792b9ff03d92b1c5022c9f
 _COMMIT_DATE=2022.05.05
 TERMUX_PKG_VERSION=${_COMMIT_DATE//./}
 TERMUX_PKG_GIT_BRANCH=master
+_TD_SRCURL=https://github.com/tdlib/td.git
+_TD_COMMIT=ab3a8282d4ee307d341071267ef1090b1a941478
 TERMUX_PKG_HOSTBUILD=true
 TERMUX_PKG_DEPENDS="readline, openssl, zlib"
 
 termux_step_post_get_source() {
-	rm -rf $TERMUX_PKG_HOSTBUILD_DIR
 	git fetch --unshallow
 	git checkout $_COMMIT
-}
 
-termux_step_post_get_source() {
-	git fetch --unshallow
-	git checkout $_COMMIT
+	rm -rf td
+	git clone $_TD_SRCURL td
+	cd td
+	git checkout $_TD_COMMIT
 }
 
 termux_step_host_build() {

